### DISCUSSION TITLE
Configure browser to only show a subset of collections

### DISCUSSION THREAD

**Discussion by hrodmn** [radiantearth/stac-browser#432] [Q&A] [answered]
I have a STAC deployed as an API with stac-fastapi-pgstac that has several distinct types of collections. I think these sets of collections would fit nicely in a set of sub-catalogs, but as discussed [here](https://github.com/stac-utils/stac-fastapi/discussions/595) I don't think it is possible to configure stac-fastapi or pgstac to do that. In order to make the collections more browsable I would like to display them in separate instances of stac-browser.

Is it possible to configure the stac-browser to only read a subset of collections by filtering on some collection-level property (e.g. `keywords`)?

**Comment by m-mohr** [comment, marked as answer]
There are two main options:

- Expose individual catalogs via stac-fastapi, this needs some code changes on your side. Then you configure STAC Browser to only read the catalogs with `apiCatalogPriority` set to `childs`.
- Alternatively, you go forward with your proposed solution and you add a `preprocessSTAC` function in the config that filters collections. But this seems like an akward hack.
